src/browser/NativeMessageInstaller.cpp

```diff
@@ -259,6 +262,35 @@ QString NativeMessageInstaller::getProxyPath() const
     return QDir::toNativeSeparators(path);
 }
 
+/** Constructs a host accessible proxy path for use with flatpak
+ *
+ * @return path Path to host accessible wrapper script (org.keepassxc.KeePassXC)
+ */
+
+#ifdef KEEPASSXC_DIST_FLATPAK
+QString NativeMessageInstaller::constructFlatpakPath() const
+{
+    // Find and extract the host flatpak data directory (in /var)
+    QString path;
+    QSettings settings("/.flatpak-info",QSettings::IniFormat);
+    settings.beginGroup("Instance");
+    QString appPath = settings.value("app-path").toString();
+
+    QRegularExpression re("^((?:/[\\.\\w-]*)+)+/app");
+    QRegularExpressionMatch match = re.match(appPath);
+    if (match.hasMatch()) {
+        // Construct a proxy path that should work with all flatpak installations
+        path = match.captured(1) + "/exports/bin/" + "org.keepassxc.KeePassXC";
+    } else {
+        // Fallback to the most common and default flatpak installation path
+        path = "/var/lib/flatpak/exports/bin/org.keepassxc.KeePassXC";
+    }
+    settings.endGroup();
+
+    return path;
+}
+#endif
```

Comment:
This should be a static free function without an external declaration if it's going to be ifdef'ed.

Reply:
Should be good now.